sechub-sereco/src/test/java/com/mercedesbenz/sechub/sereco/importer/SensitiveDataMaskingServiceTest.java

```diff
@@ -0,0 +1,206 @@
+        /* prepare */
+        String sechubConfigJson = TestFileReader.loadTextFile(SECHUB_WEBSCAN_CONFIG_FILE_WITHOUT_SENSITIVE_HEADERS);
+        SecHubConfigurationModel scanConfigWithoutSensitiveHeaders = SecHubConfiguration.createFromJSON(sechubConfigJson);
+
+        String sarifReport = TestFileReader.loadTextFile(SARIF_2_1_0_OWASP_ZAP_REPORT_FILE_WITHOUT_SENSITIVE_HEADERS);
+        SerecoMetaData data = importer.importResult(sarifReport, ScanType.WEB_SCAN);
+
+        /* execute */
+        List<SerecoVulnerability> maskedVulnerabilities = serviceToTest.maskSensitiveData(scanConfigWithoutSensitiveHeaders, data.getVulnerabilities());
+
+        /* test */
+        assertEquals(14, maskedVulnerabilities.size());
+        assertHeaderWasNotMasked("x-file-size", "123456", maskedVulnerabilities);
+        assertHeaderWasNotMasked("custom-header", "non-secret-value", maskedVulnerabilities);
+    }
+
+    @Test
+    void vulnerabilities_not_changed_when_no_web_scan_config_is_defined() throws IOException {
+        /* prepare */
+        String sarifReport = TestFileReader.loadTextFile(SARIF_2_1_0_OWASP_ZAP_REPORT_FILE_WITH_SENSITIVE_HEADERS);
+        SerecoMetaData data = importer.importResult(sarifReport, ScanType.WEB_SCAN);
+        SecHubConfigurationModel sechubConfigWithoutWebScan = new SecHubConfigurationModel();
+
+        /* execute */
+        List<SerecoVulnerability> maskedVulnerabilities = serviceToTest.maskSensitiveData(sechubConfigWithoutWebScan, data.getVulnerabilities());
+
+        /* test */
+        assertEquals(data.getVulnerabilities(), maskedVulnerabilities);
+    }
+
+    @Test
+    void vulnerabilities_not_changed_when_no_headers_are_configured() throws IOException {
+        /* prepare */
+        SecHubConfigurationModel scanConfigWithoutHeaders = new SecHubConfigurationModel();
+        scanConfigWithoutHeaders.setWebScan(new SecHubWebScanConfiguration());
+
+        String sarifReport = TestFileReader.loadTextFile(SARIF_2_1_0_OWASP_ZAP_REPORT_FILE_WITH_SENSITIVE_HEADERS);
+        SerecoMetaData data = importer.importResult(sarifReport, ScanType.WEB_SCAN);
+
+        /* execute */
+        List<SerecoVulnerability> maskedVulnerabilities = serviceToTest.maskSensitiveData(scanConfigWithoutHeaders, data.getVulnerabilities());
+
+        /* test */
+        assertEquals(data.getVulnerabilities(), maskedVulnerabilities);
+    }
+
+    @Test
+    void empty_sarif_report_with_no_findings_handled_without_changing_anything() throws IOException {
+        /* prepare */
+        String sechubConfigJson = TestFileReader.loadTextFile(SECHUB_WEBSCAN_CONFIG_FILE_WITH_SENSITIVE_HEADERS);
+        SecHubConfigurationModel scanConfig = SecHubConfiguration.createFromJSON(sechubConfigJson);
+
+        String sarifReport = TestFileReader.loadTextFile(SARIF_2_1_0_OWASP_ZAP_EMPTY_REPORT_FILE);
+        SerecoMetaData data = importer.importResult(sarifReport, ScanType.WEB_SCAN);
```

Comment:
Please create a method like my example below and use it inside all of your tests (makes it easier to read, understand and less boilerplate code)

```java
SerecoMetaData createTestMetaDataFromProductResult(String path){
        String productResult = TestFileReader.loadTextFile(path);
        SerecoMetaData data = importer.importResult(productResult, ScanType.WEB_SCAN);
        return data;
}
```